frame/elections/src/lib.rs

```diff
@@ -904,24 +990,23 @@ impl<T: Config> Pallet<T> {
 		candidates_and_deposit.append(&mut Self::implicit_candidates_with_deposit());
 
 		if candidates_and_deposit.len().is_zero() {
-			Self::deposit_event(Event::EmptyTerm);
-			return T::DbWeight::get().reads(3)
+			return Err(Error::EmptyTerm)
 		}
 
-		// All of the new winners that come out of phragmen will thus have a deposit recorded.
+		// All of the new winners that come out of the election will thus have a deposit recorded.
 		let candidate_ids =
 			candidates_and_deposit.iter().map(|(x, _)| x).cloned().collect::<Vec<_>>();
 
 		// helper closures to deal with balance/stake.
 		let total_issuance = T::Currency::total_issuance();
 		let to_votes = |b: BalanceOf<T>| T::CurrencyToVote::to_vote(b, total_issuance);
-		let to_balance = |e: ExtendedBalance| T::CurrencyToVote::to_currency(e, total_issuance);
 
 		let mut num_edges: u32 = 0;
 
 		let max_voters = <T as Config>::MaxVoters::get() as usize;
 		// used for prime election.
 		let mut voters_and_stakes = Vec::new();
+
 		match Voting::<T>::iter().try_for_each(|(voter, Voter { stake, votes, .. })| {
```

Comment:
The match statement was only here because we wanted to log something on error. Now it is pretty redundant and can be replaced with `things.try_for_each(|..| ..)?`.

Reply:
Good catch, that's right!